src/app/gui/queryresults/queryresultsservice.js

```diff
@@ -94,23 +94,37 @@ function QueryResultsService() {
     mapcontrol: null, // add current toggled map control if toggled
     toggleeventhandler: null
   };
+
   this.setters = {
+
     /**
-     * Method call when response is handled by Data Provider
+     * Hook method called when response is handled by Data Provider
+     * 
      * @param queryResponse
-     * @param options: add is used to know if is a new query request or add/remove query request
+     * @param {{ add: boolean }} options `add` is used to know if is a new query request or add/remove query request
      */
-    setQueryResponse(queryResponse, options={add:false}) {
-      const {add} = options;
-      // in case of new request results reset the query otherwise maintain the previous request
-      if (!add) {
+    setQueryResponse(queryResponse, options = { add: false }) {
+
+      // set mandatory queryResponse fields
+      if (!queryResponse.data)  queryResponse.data    = [];
+      if (!queryResponse.query) queryResponse.query   = { external: { add: false, filter: {SELECTED: false }}};
+      if (!queryResponse.query.external) queryResponse.query.external = { add: false, filter: {SELECTED: false }};
```

Comment:
In the future we might consider rewriting it to something like this:

```js
// set mandatory queryResponse fields
queryResponse = utils.merge({
  data: [],
  query: {
    external: {
      add: false,
      filter: { SELECTED: false }
    }
  }
}, queryResponse);
```

but that's okay for now 👍